### ISSUE TITLE
facing issues while upgrading from v0.7.2 to v0.8 and above

### ISSUE THREAD

**Issue by vinayus** [jcmoraisjr/haproxy-ingress#517]
i have been facing issue with upgrading the images to v0.8 and above.

this is my yaml file: https://pastebin.com/XsMEkNwp

same yaml file from v0.7.2 to v0.7.5 works fine and from v0.8 onwards I'm unable to bring it up.

haproxy stats page and logs isn't helping me identify.

FYI: I can only access stats page and I can't access the default backend from v0.8 onwards.
Kindly help

**Comment by jcmoraisjr** [comment]
Hi, thanks for evaluating v0.8. The problem is deploying the controller itself or exposing your app? What is the problem?

If the former, try starting with [this yaml](https://haproxy-ingress.github.io/resources/haproxy-ingress.yaml) instead, and note that the service you created to expose haproxy ingress is exposing port `443` twice.

If the later please paste some of your ingress objects.

**Comment by vinayus** [comment]
hi, thank you for responding so quickly.
i don't have issue with deploying the controller, it deploys fine and doesn't print any error messages.

When I try to access the controller(say on 443 via nodePort or daemonset) to reach out to my default backend, all I get is empty response or connection closed from the browser.

The same yaml when used with 0.7.2 or 0.7.5 image works fine.

Below is the log extract from the v0.10 tag. Pretty much everything from v0.8, i'm with this issue.

`I0203 10:56:58.464353       6 launch.go:177]
Name:       HAProxy
Release:    v0.10-snapshot.2
Build:      git-4adaeb8
Repository: https://github.com/jcmoraisjr/haproxy-ingress
I0203 10:56:58.464692       6 launch.go:180] Watching for ingress class: haproxy
I0203 10:56:58.466113       6 launch.go:397] Creating API client for https://10.96.0.1:443
I0203 10:56:58.505693       6 launch.go:409] Running in Kubernetes Cluster version v1.15 (v1.15.9) - git (clean) commit 2e808b7cb054ee242b68e62455323aa783991f03 - platform linux/amd64
I0203 10:56:58.513880       6 launch.go:201] validated ingress-controller/ingress-default-backend as the default backend
I0203 10:56:58.671374       6 controller.go:164] using auto generated fake certificate
I0203 10:56:59.550637       6 controller.go:231] starting Ingress controller
I0203 10:56:59.551010       6 controller.go:263] Starting HAProxy update id=1
I0203 10:56:59.552281       6 leaderelection.go:241] attempting to acquire leader lease  ingress-controller/ingress-controller-leader-haproxy...
I0203 10:56:59.569340       6 leaderelection.go:251] successfully acquired lease ingress-controller/ingress-controller-leader-haproxy
I0203 10:56:59.569608       6 status.go:171] new leader elected: haproxy-ingress-6f795b55c-77p86
I0203 10:56:59.647664       6 instance.go:318] HAProxy successfully reloaded
I0203 10:56:59.647777       6 controller.go:314] Finish HAProxy update id=1: parse_ingress=0.686399ms write_tmpl=3.503935ms reload_haproxy=92.367668ms total=96.558002ms
I0203 10:57:01.551225       6 controller.go:263] Starting HAProxy update id=2
I0203 10:57:01.551979       6 instance.go:304] old and new configurations match
I0203 10:57:01.552031       6 controller.go:314] Finish HAProxy update id=2: parse_ingress=0.425210ms total=0.425210ms
I0203 10:57:19.528955       6 controller.go:263] Starting HAProxy update id=3
I0203 10:57:19.551923       6 instance.go:301] HAProxy updated without needing to reload. Commands sent: 3
I0203 10:57:19.552058       6 controller.go:314] Finish HAProxy update id=3: parse_ingress=1.245018ms write_tmpl=21.624045ms total=22.869063ms`



To summarise: controller's stat page(1024 port) is the only endpoint I can access from browser
both 80 and 443 aren't accessible.


PS: 443 was a typo. Sorry about that.

**Comment by vinayus** [comment]
here is my very simple ingress conf..

```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    run: ingress-default-backend
  name: ingress-default-backend
  namespace: ingress-controller
spec:
  selector:
    matchLabels:
      run: ingress-default-backend
  template:
    metadata:
      labels:
        run: ingress-default-backend
    spec:
      containers:
      - name: ingress-default-backend
        image: gcr.io/google_containers/defaultbackend:1.0
        ports:
        - containerPort: 8080
        resources:
          limits:
            cpu: 10m
            memory: 20Mi
---
apiVersion: v1
kind: Service
metadata:
  name: ingress-default-backend
  namespace: ingress-controller
spec:
  ports:
  - port: 8080
  selector:
    run: ingress-default-backend
---
```

**Comment by jcmoraisjr** [comment]
Hi, you're facing the same problem that #516 should solve. If you deploy any ingress object the frontend will be created, provided that the ingress has the `host` configuration. A new v0.8 with that PR will be tagged shortly.

**Comment by vinayus** [comment]
Thank you. Please cherry pick the same to v0.10 as well if possible.

**Comment by jcmoraisjr** [comment]
Hi, v0.8.1, v0.9-beta.2 and v0.10-snapshot.3 now fixes this issue. Closing. Please have a look and let me know if you still have any problem.

**Comment by vinayus** [comment]
Just tried and they are working fine with `v0.10-snapshot.3`
